decklang/decklang.ne

```diff
@@ -1,11 +1,35 @@
+@builtin "string.ne"
+@builtin "number.ne"
+@builtin "whitespace.ne"
 
-@include "plugins/text.ne"
-@include "plugins/font.ne"
-@include "primitives_advanced.ne"
-@include "primitives.ne"
+Program -> Expressions | null
+Expressions -> Expression Expressions
+Expression -> Loop | VariableDeclaration | Directive | Comment | NewLine
 
-main -> _ (line_of_code _):*
+Comment -> "`" btstring
+Directive -> "text" _ "=" _ ArgumentList | "font" _ "=" _ ArgumentList
+VariableDeclaration -> Variable _ "=" _ ArgumentList
+Loop -> LoopStart LoopBody
 
-line_of_code ->
-  text
-| font
\ No newline at end of file
+LoopStart -> "loop" _ "=" _ "<" _ Assignment _ "> to " LoopGoal
+LoopGoal -> Integer | Variable | LoopVariable
+LoopBody -> LoopEnd | Expression LoopBody
+LoopEnd -> "endloop"
+
+ArgumentList -> Argument | Argument _ "," _ ArgumentList
+Argument -> Identifier | String | Integer | HexString | Variable | LoopVariable
+Assignment -> Identifier _ "=" _ Argument
+
+Variable -> "[" _ Identifier _ "]"
```

Comment:
no need to have this in the grammar, they're pulled out and replaced before seen by the program